transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringEventLoop.java

```diff
@@ -104,19 +104,25 @@ void add(AbstractIOUringChannel ch) {
         int fd = ch.socket.intValue();
 
         channels.put(fd, ch);
+        ringBuffer.ioUringSubmissionQueue().incrementHandledFds();
```

Comment:
I think you don't want to increment here in the case that `channels.put` returns non-null, right?

Reply:
I think I actually want as I also always decrement... below. I am missing something ?

Reply:
Maybe this isn't possible for other sequencing reasons, but what if the same fd is replaced 3 times in succession.. then count will be +3. Then when removed it will only be -1 since it's only decremented once (if called multiple times only the first will decrement). So we end up with no fd but still a positive count. I could also be missing something :)

Reply:
Thus not possible

Reply:
Ok maybe the multiple case is not possible but you are saying that the `put` will never overwrite? If so what is the purpose of this check after `remove` https://github.com/netty/netty/pull/10625/files#diff-edc47287cb4c12155b5456eb05260bcbR118-R125?

Reply:
You are right ... I guess I should only increment here if null is returned and only decrement if we not put back the channel in the remove method ... WDYt ?

Reply:
Yes I think if you just increment here if null is returned from `put` then the decrement can stay as you have it